File: app/src/main/java/org/tvheadend/tvhclient/data/repository/AppRepository.java
package org.tvheadend.tvhclient.data.repository;

import org.tvheadend.tvhclient.data.source.ChannelData;
import org.tvheadend.tvhclient.data.source.ChannelTagData;
import org.tvheadend.tvhclient.data.source.ConnectionData;
import org.tvheadend.tvhclient.data.source.MiscData;
import org.tvheadend.tvhclient.data.source.ProgramData;
import org.tvheadend.tvhclient.data.source.RecordingData;
import org.tvheadend.tvhclient.data.source.SeriesRecordingData;
import org.tvheadend.tvhclient.data.source.ServerProfileData;
import org.tvheadend.tvhclient.data.source.ServerStatusData;
import org.tvheadend.tvhclient.data.source.TagAndChannelData;
import org.tvheadend.tvhclient.data.source.TimerRecordingData;

import javax.inject.Inject;

public class AppRepository implements RepositoryInterface {

    private final ChannelData channelData;
    private final ProgramData programData;
    private final RecordingData recordingData;
    private final SeriesRecordingData seriesRecordingData;
    private final TimerRecordingData timerRecordingData;
    private final ConnectionData connectionData;
    private final ChannelTagData channelTagData;
    private final ServerStatusData serverStatusData;
    private final ServerProfileData serverProfileData;
    private final TagAndChannelData tagAndChannelData;
    private final MiscData miscData;

    @Inject
    public AppRepository(ChannelData channelData,
                         ProgramData programData,
                         RecordingData recordingData,
                         SeriesRecordingData seriesRecordingData,
                         TimerRecordingData timerRecordingData,
                         ConnectionData connectionData,
                         ChannelTagData channelTagData,
                         ServerStatusData serverStatusData,
                         ServerProfileData serverProfileData,
                         TagAndChannelData tagAndChannelData,
                         MiscData miscData) {
        this.channelData = channelData;
        this.programData = programData;
        this.recordingData = recordingData;
        this.seriesRecordingData = seriesRecordingData;
        this.timerRecordingData = timerRecordingData;
        this.connectionData = connectionData;
        this.channelTagData = channelTagData;
        this.serverStatusData = serverStatusData;
        this.serverProfileData = serverProfileData;
        this.tagAndChannelData = tagAndChannelData;
        this.miscData = miscData;
    }

    @Override
    public TimerRecordingData getTimerRecordingData() {
        return timerRecordingData;
    }

    @Override
    public SeriesRecordingData getSeriesRecordingData() {
        return seriesRecordingData;
    }

    @Override
    public RecordingData getRecordingData() {
        return recordingData;
    }

    @Override
    public ChannelData getChannelData() {
        return channelData;
    }

    @Override
    public ChannelTagData getChannelTagData() {
        return channelTagData;
    }

    @Override
    public ProgramData getProgramData() {
        return programData;
    }

    @Override
    public ConnectionData getConnectionData() {
        return connectionData;
    }

    @Override
    public ServerStatusData getServerStatusData() {
        return serverStatusData;
    }

    @Override
    public ServerProfileData getServerProfileData() {
        return serverProfileData;
    }

    @Override
    public TagAndChannelData getTagAndChannelData() {
        return tagAndChannelData;
    }

    @Override
    public MiscData getMiscData() {
        return miscData;
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/data/repository/AppRepository.kt
package org.tvheadend.tvhclient.data.repository

import android.content.Intent
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import org.tvheadend.tvhclient.data.source.*
import org.tvheadend.tvhclient.ui.common.Event
import org.tvheadend.tvhclient.ui.common.NetworkStatus
import javax.inject.Inject

class AppRepository @Inject
constructor(
        override val channelData: ChannelDataSource,
        override val programData: ProgramDataSource,
        override val recordingData: RecordingDataSource,
        override val seriesRecordingData: SeriesRecordingDataSource,
        override val timerRecordingData: TimerRecordingDataSource,
        override val connectionData: ConnectionDataSource,
        override val channelTagData: ChannelTagDataSource,
        override val serverStatusData: ServerStatusDataSource,
        override val serverProfileData: ServerProfileDataSource,
        override val tagAndChannelData: TagAndChannelDataSource,
        override val miscData: MiscDataSource,
        override val subscriptionData: SubscriptionDataSource,
        override val inputData: InputDataSource
) : RepositoryInterface {

    private var isUnlocked = MutableLiveData<Boolean>()
    private var snackbarMessage = MutableLiveData<Event<Intent>>()
    private var networkStatus = MutableLiveData<NetworkStatus>()
    private var connectionToServerAvailable = MutableLiveData<Boolean>()

    init {
        isUnlocked.value = false
        networkStatus.value = NetworkStatus.NETWORK_UNKNOWN
        connectionToServerAvailable.value = false
    }

    fun getSnackbarMessage(): LiveData<Event<Intent>> = snackbarMessage

    fun setSnackbarMessage(msg: Intent) {
        snackbarMessage.value = Event(msg)
    }

    fun getNetworkStatus(): LiveData<NetworkStatus> = networkStatus

    fun setNetworkStatus(status: NetworkStatus) {
        networkStatus.value = status
    }

    fun getIsUnlocked(): LiveData<Boolean> = isUnlocked

    fun setIsUnlocked(unlocked: Boolean) {
        isUnlocked.value = unlocked
    }

    fun getConnectionToServerAvailable(): LiveData<Boolean> = connectionToServerAvailable

    fun setConnectionToServerAvailable(available: Boolean) {
        connectionToServerAvailable.value = available
    }
}
